diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt

```diff
@@ -296,6 +306,64 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
                 }
     }
 
+    /**
+     * Checks that members of [VALUE_PARAMETER_LIST] (list of function parameters at declaration site) are separated with newlines.
+     * Also checks that entries of [SUPER_TYPE_LIST] are separated by newlines.
+     */
+    private fun handleList(node: ASTNode) {
+        val (numEntries, entryType) = when (node.elementType) {
+            VALUE_PARAMETER_LIST -> (node.psi as KtParameterList).parameters.size to "value parameters"
+            SUPER_TYPE_LIST -> (node.psi as KtSuperTypeList).entries.size to "supertype list entries"
+            else -> {
+                log.warn("Unexpected node element type ${node.elementType}")
+                return
+            }
+        }
+        if (numEntries > configuration.maxParametersInOneLine) {
+            when (node.elementType) {
+                VALUE_PARAMETER_LIST -> handleFirstValueParameter(node)
+            }
+
+            node
+                .children()
+                .filter {
+                    it.elementType == COMMA &&
+                        !it.treeNext.run { elementType == WHITE_SPACE && textContains('\n') }
+                }
+                .toList()
+                .onlyIf({ isNotEmpty() }) { invalidCommas ->
+                    WRONG_NEWLINES.warnAndFix(configRules, emitWarn, isFixMode,
+                        "$entryType should be placed on different lines in declaration of <${node.getParentIdentifier()}>", node.startOffset, node) {
+                        invalidCommas.forEach { comma ->
+                            val nextWhiteSpace = comma.treeNext.takeIf { it.elementType == WHITE_SPACE }
+                            comma.appendNewlineMergingWhiteSpace(nextWhiteSpace, nextWhiteSpace?.treeNext ?: comma.treeNext)
+                        }
+                    }
+                }
+        }
+    }
+
+    private fun handleFirstValueParameter(node: ASTNode) {
+        node
+            .children()
+            .takeWhile { !it.textContains('\n') }
+            .filter { it.elementType == VALUE_PARAMETER }
+            .toList()
+            .onlyIf({ size > 1 }) {
+                WRONG_NEWLINES.warnAndFix(configRules, emitWarn, isFixMode, "first parameter should be placed on a separate line " +
+                        "or all other parameters should be aligned with it in declaration of <${node.getParentIdentifier()}>", node.startOffset, node) {
+                    node.appendNewlineMergingWhiteSpace(it.first().treePrev.takeIf { it.elementType == WHITE_SPACE }, it.first())
+                }
+            }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun ASTNode.getParentIdentifier() = when (treeParent.elementType) {
+        PRIMARY_CONSTRUCTOR -> treeParent.treeParent
+        SECONDARY_CONSTRUCTOR -> parent(CLASS)!!
+        else -> treeParent
+    }
+        .getIdentifierName()?.text
```

Comment:
I found that It returns `null` when there is lambda.

Reply:
Where?

Reply:
Oh, I see, rule doesn't distinguish functional types from function declarations. Good point, I'll fix it.